### ISSUE TITLE
Add support for "click-though" (mouse activity captured by an unfocused window) for desktop applications on MacOS.

### ISSUE THREAD

**Issue by OKGougou** [godotengine/godot-proposals#8711]
### Describe the project you are working on

[SpeakerView](https://github.com/GRIS-UdeM/SpeakerView) is the 3D view (made with Godot) of a concert hall's speaker configuration for [SpatGRIS](https://github.com/GRIS-UdeM/SpatGRIS) (a sound spatilization tool).

![SpeakerView](https://github.com/godotengine/godot-proposals/assets/75331877/7ca07313-30fe-4d74-b864-7eaca570023f)

### Describe the problem or limitation you are having in your project

SpatGRIS and SpeakerView form an inseparable pair. Users often switch from one window to the other when working on a project. When the SpeakerView window is not in focus, the first mouse click in the window is an input event under Linux and Windows, but not under MacOS. The user must click in the window a second time to interact with SpeakerView.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Godot can be given click-though support on MacOS by adding the instance method [acceptsFirstMouse(for:)] (https://developer.apple.com/documentation/appkit/nsview/1483410-acceptsfirstmouse) to the source code and recompiling it and the export templates.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

In the file platform/macos/godot_content_view.mm, add the method:
```
- (BOOL)acceptsFirstMouse:(NSEvent *)event {
    return YES;
}
```

Not sure if it should go under the Focus or Mouse section...

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, this is a core feature for input events on MacOS.

### Is there a reason why this should be core and not an add-on in the asset library?

This change would allow desktop applications created with Godot to act in the same way on Linux, Windows and MacOS.
I don't think an add-on would solve this problem.

**Comment by Calinou** [comment]
Should this be done by default? This would better match Windows/Linux behavior and lead to fewer surprises when exporting a project across platforms, but at the same time, I think most macOS apps have this disabled by default.

**Comment by OKGougou** [comment]
Indeed, this is not the default behavior of most MacOS applications. But it seems that a few components are chosen to respond to click-through, such as buttons, sliders or toggles.
I know that [JUCE implements it](https://github.com/juce-framework/JUCE/blob/d054f0d14dcac387aebda44ce5d792b5e7a625b3/modules/juce_gui_basics/native/juce_NSViewComponentPeer_mac.mm#L2109).
[Firefox](https://searchfox.org/mozilla-central/source/widget/cocoa/nsChildView.mm#2334) and [Qt](https://github.com/qt/qtbase/blob/39290c508e2c7fa816c6f79b3eb284adc0cad3c0/src/plugins/platforms/cocoa/qnsview_mouse.mm#L271) under certain conditions.
Maybe it would be better to have an option in the Godot project settings?
